packages/grid/_modules_/grid/hooks/features/clipboard/useGridClipboard.ts

```diff
@@ -82,7 +82,7 @@ export const useGridClipboard = (apiRef: GridApiRef): void => {
     [apiRef],
   );
 
-  useGridApiEventHandler(apiRef, GRID_KEYDOWN, handleKeydown);
+  useGridApiEventHandler(apiRef, GridEvents.keydown, handleKeydown);
```

Comment:
I'm fine with not adding native events to the enum.

```suggestion
  useGridApiEventHandler(apiRef, 'keydown', handleKeydown);
```

This `keydown` is only used in this hook and here:

https://github.com/mui-org/material-ui-x/blob/09b936fd34d9c36b92b24fbdacd6cc4b0d45186c/packages/grid/_modules_/grid/hooks/root/useEvents.ts#L99-L104

Reply:
AFAIK, it's not a native event here, not as in https://github.com/mui-org/material-ui-x/pull/2279#discussion_r684542859

Reply:
This event handler is weird. We listen for the native event to publish a grid's event. For the scroll this doesn't happen (example below), the native listener is added in the same hook where it's used. I propose we don't add the native event to the enum and in a second iteration we move the `useNativeEventListener` to the clipboard hook.

https://github.com/mui-org/material-ui-x/blob/09b936fd34d9c36b92b24fbdacd6cc4b0d45186c/packages/grid/_modules_/grid/hooks/features/virtualization/useGridVirtualRows.ts#L387

Reply:
@m4theushw which events would you remove from the enum ? 
"keydown" or are there others ?

Reply:
Yeah, just remove "keydown". It's already marked as ignored so no problem removing it.

Reply:
But their is a listener on it

```ts
useGridApiEventHandler(apiRef, GridEvents.keydown, handleKeydown);
```

You would replace it like this ?

```ts
useGridApiEventHandler(apiRef, 'keydown', handleKeydown);
```

Reply:
Yes, I would leave `'keydown'`. The scroll listener is also hardcoded.

Reply:
For the scroll it's using`useNativeEventListener` not `useGridApiEventListener`

```ts
  useNativeEventListener(apiRef, windowRef, 'scroll', handleScroll, { passive: true });
```

Reply:
> By the way, we never publishes publishEvent(GridEvents.keydown, ...) or publishEvent('keydown', ...

That's the problem I was mentioning in https://github.com/mui-org/material-ui-x/pull/2279#discussion_r685487183. We add the native listener to publish a grid's event.

> Is this useGridApiEventHandler(apiRef, GridEvents.keydown, handleKeydown); even called ?

It's used for the clipboard feature.

Reply:
https://github.com/mui-org/material-ui-x/pull/2279/commits/ee602b614b38d300aad2b2d4971eade5fe03208b

I'm now using directly `useNativeEventListener` in the clipboard feature to totally remove the event from the grid events structure :+1: